src/main/java/io/github/nucleuspowered/nucleus/modules/playerinfo/commands/NearCommand.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Permissions(suggestedLevel = SuggestedLevel.USER, supportsOthers = true)
+@RegisterCommand("near")
+@EssentialsEquivalent("near")
+@NonnullByDefault
+public class NearCommand extends AbstractCommand<CommandSource> implements Reloadable {
+
+    private final String radiusKey = "radius";
+    private int maxRadius;
+
+    @Override
+    protected Map<String, PermissionInformation> permissionSuffixesToRegister() {
+        final Map<String, PermissionInformation> pi = super.permissionSuffixesToRegister();
+        pi.put("maxexempt", PermissionInformation.getWithTranslation("permission.near.maxexempt", SuggestedLevel.MOD));
+        return pi;
+    }
+
+    @Override
+    public CommandElement[] getArguments() {
+        return new CommandElement[] {
+                GenericArguments.optionalWeak(requirePermissionArg(GenericArguments.user(Text.of(NucleusParameters.ONE_USER_PLAYER_KEY)), this.permissions.getOthers())),
+                GenericArguments.optionalWeak(GenericArguments.integer(Text.of(this.radiusKey)))
+        };
+    }
+
+    //near radius
+    //near player radius
+    @Override
+    protected CommandResult executeCommand(CommandSource src, CommandContext args, Cause cause) throws Exception {
+        final User user = this.getUserFromArgs(User.class, src, NucleusParameters.Keys.PLAYER, args);
+        final Location<World> location;
+        if (user.isOnline()) {
+            location = user.getPlayer().get().getLocation();
+        } else {
+            location = Nucleus.getNucleus().getUserDataManager().get(user).flatMap(x -> x.get(CoreUserDataModule.class).getLogoutLocation()).orElseThrow(() -> ReturnMessageException.fromKey("command.near.location.nolocation", user.getName()));
+        }
+
+        int radius = maxRadius;
+        final Optional<Integer> radiusOpt = args.getOne(radiusKey);
+        if(radiusOpt.isPresent()) {
+            int inputRadius = radiusOpt.get();
+            //Check if executer has max radius override permission
+            if (inputRadius > maxRadius && this.permissions.testSuffix(src, "maxexempt")) {
+                radius = inputRadius;
+            } else {
+                src.sendMessage(Nucleus.getNucleus().getMessageProvider().getTextMessageWithFormat("command.near.radiustoobig", this.maxRadius));
+            }
+        }
+
+        final PlayerOnlineService playerOnlineService = getServiceManager().getServiceUnchecked(PlayerOnlineService.class);
+        final List<Player> nearbyPlayers = location.getExtent().getNearbyEntities(location.getPosition(), radius).stream()
+                .filter(Player.class::isInstance)
+                .map(Player.class::cast)
+                .filter(e -> e.getUniqueId() != user.getUniqueId() && playerOnlineService.isOnline(src, e))
+                .sorted((o1, o2) -> (int) Math.abs(o1.getLocation().getPosition().distance(o2.getPosition())))//Sort by distance
```

Comment:
One thing - it will be a lot more performant to use `distanceSquared` and not `abs` it. You can keep it as a double too, the cast is unnecessary.

Reply:
The cast is needed as compare is expected to return a int

Reply:
Oh, yes, it does. I'd prefer to do something about this and sort by the float, but this will do for now, I guess.